content/post/how-to-be-bleeding-edge-c++-proposals.md

```diff
@@ -0,0 +1,98 @@
++++
+categories = []
+date = "2015-09-05T23:57:10-07:00"
+description = ""
+keywords = []
+title = "How to be bleeding edge – C++ proposals that'll change the way you code"
+
++++
+
+In this article, I'll go over a few of the most recently implemented C++ features and extensions in GCC and Clang. Some might just change the face of your code forever. At the very least, demonstrating your bleeding edge knowledge in interviews will be sure to get you some points.
+
+Initialized lambda captures
+===========================
+
+This is a C++14 feature that as far as I've been concerned means "capture by move" support. As a stickler for optimization, I hate unnecessary copies. With initialized lambda captures, I can be even more of a stickler:
+
+```
+std::vector<Things> largeVector;
+// fill largeVector with lots of stuff
+auto result = std::async([largeVector{std::move(largeVector)}] {
+    // process largeVector
+});
+```
+
+The proposal: [N3648](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3648.html)
+
+Digit separator
+===============
+
+Here's another C++14 feature that you may find nice every now and then. For numeric literals, you can now insert a separator:
+
+```
+namespace {
+    constexpr int kSecondsInAWeek = 604'800;
+}
+```
+
+Unfortunately, this'll probably break your syntax highligher (as you can see above).
+
+The proposal: [N3781](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3781.pdf)
+
+Literal operator templates
+==========================
+
+With user defined literals, we can do some pretty handy things with strings at compile time. But the standard's missing something: A template form of the literal operator for characters and strings. So with the current standard, we can't change the type of the operator based on the contents of the string.
+
+One use case is for a type-safe formatter:
+
+```
+template<typename CharT, CharT... String>
+auto operator""_format() { return Formatter<String...>(); }
+
+void foo() {
+    auto str = "Hello %s!\n"_format("world"); // okay!
+    auto str = "Hello %s!\n"_format(1); // compile time error!
+}
+```
```

Comment:
Never thought of that use case. Very cool.

I don't know if your goal is to provide syntactically correct code or just provide examples, but if the code is to be syntactically correct the arguments to the `operator""_format` function would have to be `const char*`. I personally think it's better to have code that is syntactically correct so long as doing so doesn't interfere with the demonstration, but in this case I think it could get verbose. I would probably just put `...` for the function arguments and return `Formatter<String>(...)`